### DISCUSSION TITLE
Combobox with displayValue and placeholder

### DISCUSSION THREAD

**Discussion by thraizz** [tailwindlabs/headlessui#1533] [Help] [answered]
Hi, I am trying to implement a Combobox that has both a function for displaying an objects property as value and a placeholder. This seems to be not possible, my approach was to return undefined in the `displayValue` function when no object is passed in, e.g.:
```jsx
<Combobox.Input
  displayValue={(myObject) =>
    myObject ? myObject.name : undefined
  }
  placeholder="My placeholder"
  // [...]
/>
```

Unfortunately, this does not seem to work. Returning the placeholder in the displayValue function does not work either, as my styling depends on the presence of a placeholder.

Is there something I am missing here? Any hint would be greatly appreciated!

**Comment by thraizz** [comment, marked as answer]
Nevermind, I solved this myself: You can just return an empty string in the displayValue function, the Combobox.Input will then show the placeholder instead.
So something like:
```jsx
<Combobox.Input
    displayValue={(myObject) =>
      myObject ? myObject.name : ""
    }
    placeholder="My placeholder"
    // [...}
/>
```
should work.

**Reply by toniopelo** [reply]
Thanks @thraizz. For those who would end up here like me, this is not working on `1.6.4`, so it probably as been fixed in `1.6.5` or `1.6.6`, just upgrade your package :).
